.github/workflows/test-and-build.yaml

```diff
@@ -26,34 +26,27 @@ jobs:
     needs: test
     runs-on: ubuntu-latest
     steps:
-      -
-        name: Checkout
+      - name: Checkout
         uses: actions/checkout@v2
-      -
-        name: Get the tag name
+      - name: Get the tag name
         run: |
           echo "TAG=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
-      -
-        name: Docker meta
+      - name: Docker meta
         id: meta
         uses: docker/metadata-action@v3
         with:
-          images: pihole/pihole
+          images: ${{ secrets.DOCKERHUB_USER }}/pihole
```

Comment:
May need to make this another secret in the case that namespace and user do not match.

Reply:
<html><body>
<!--StartFragment-->

GITHUB_ACTOR | The name of the person or app that initiated the workflow. For example, octocat.
-- | --
GITHUB_REPOSITORY | The owner and repository name. For example, octocat/Hello-World.

<!--EndFragment-->
</body>
</html>

Reply:
Possibly something [like this](https://github.com/pi-hole/FTL/blob/development/.github/workflows/build.yml#L50-L59)

Use like [this](https://github.com/pi-hole/FTL/blob/development/.github/workflows/build.yml#L92)

I think it is best to use custom secrets in the case of dockerhub. Using `github.repository`  in our case would be wrong, because here it is `pi-hole`, whereas on dockerhub it is `pihole`

Reply:
``` yml
-       name: Determine Docker Hub Namespace
        id: namespace
        run: |
          DOCKERHUB_USER=${{ secrets.DOCKERHUB_USER }}
          DOCKERHUB_NAMESPACE=${{ secrets.DOCKERHUB_NAMESPACE }}          
          echo ::set-output name=DOCKERHUB_NAMESPACE::${DOCKERHUB_NAMESPACE:=${DOCKERHUB_USER}}

-       name: Docker meta
        id: meta
        uses: docker/metadata-action@v3
        with:
          images: ${{ steps.namespace.outputs.DOCKERHUB_NAMESPACE }}/pihole
```

etc etc

Reply:
That looks a bit hacky. Probably best to just have `secrets.DOCKERHUB_NAMESPACE`. Forks will still fail until that variable is set but at least it's configurable now.

Reply:
Yeah, indeed - if someone is going to need to add in `DOCKERHUB_USER` anyway, no point in having the check!